pkg/catalog/catalog.go

```diff
@@ -129,8 +131,67 @@ func getDefaultBuilderImages(client *occlient.Client) ([]CatalogImage, error) {
 		imageStreamTagMap[imageStreamTag.Name] = imageStreamTag
 	}
 
-	var builderImages []CatalogImage
+	builderImages := getBuildersFromImageStreams(imageStreams, imageStreamTagMap)
+
+	return builderImages, nil
+}
 
+// SpliceSupportedTags splits the tags in to fully supported and unsupported tags
+func SpliceSupportedTags(catalogImage CatalogImage) ([]string, []string) {
+
+	var supTag, nonSupTag []string
+	tagMap := createImageMap(catalogImage.imageStreamRef.Spec.Tags)
+	for _, tag := range catalogImage.NonHiddenTags {
+		imageName := tagMap[tag]
+		if isSupportedImage(imageName) {
+			supTag = append(supTag, tag)
+		} else {
+			nonSupTag = append(nonSupTag, tag)
+		}
+	}
+	return supTag, nonSupTag
+}
+
+func createImageMap(tagRefs []imagev1.TagReference) map[string]string {
+	tagMap := make(map[string]string)
+	for _, tagRef := range tagRefs {
+		// TODO: the From here can be of multiple types so we need to handle that
+		imageName := tagRef.From.Name
+		if tagRef.From.Kind == "DockerImage" {
+			urlImageName := strings.SplitN(tagRef.From.Name, "/", 2)[1]
+			ns, img, _, _, _ := occlient.ParseImageName(urlImageName)
+			imageName = ns + "/" + img
+		} else if tagRef.From.Kind == "ImageStreamTag" {
+			tagList := strings.Split(tagRef.From.Name, ":")
+			tag := tagList[len(tagList)-1]
+			// if the kind is a image stream tag that means its pointing to an existing dockerImage or image stream image
+			// we just look it up from the tapMap we already have
+			imageName = tagMap[tag]
+		}
+		tagMap[tagRef.Name] = imageName
+	}
+	return tagMap
+}
+
+func isSupportedImage(imgName string) bool {
+	supportedImages := []string{
+		"redhat-openjdk-18/openjdk18-openshift",
+		"openjdk/openjdk-11-rhel8",
+		"openjdk/openjdk-11-rhel7",
+		"rhscl/nodejs-8-rhel7",
+		"rhoar-nodejs/nodejs-8",
+		"rhoar-nodejs/nodejs-10",
+	}
+	for _, supImage := range supportedImages {
+		if supImage == imgName {
+			return true
+		}
+	}
+	return false
+}
+
+func getBuildersFromImageStreams(imageStreams []imagev1.ImageStream, ism map[string]imagev1.ImageStreamTag) []CatalogImage {
```

Comment:
Again, I know it's not an exported function but, it's pretty large and some comment would be really helpful, IMO.

Reply:
added